src/main/scala/arcana/world/ArgumentEvaluator.scala

```diff
@@ -0,0 +1,41 @@
+package arcana.world
+
+import scala.util.Try
+
+class ArgumentEvaluator {
+  def evaluat(args: Array[String]): Unit = {
+    import ArgumentEvaluator._
+
+    var needTypeInfo: Boolean = false // TODO add more judge flag
+    args.foreach((arg: String) => {
+      if (availableOptions.contains(arg)) {
+        arg match {
+          case "--help" => {
+            printHelpHint
+            return
+          }
+          case "--type" => needTypeInfo = true // TODO add more options
+        }
+      } else {
+        throw ArgumentException(
+          arg + " is not an available option.\n" +
+            "type arcana --help for more infomation.\n"
+        )
+      }
+    })
+
+    new Repl().start(new ReplOptions(needTypeInfo))
```

Comment:
Instantiation of case classes need no `new`.
```suggestion
    new Repl().start(ReplOptions(needTypeInfo))
```

Reply:
And do not start repl here. Later we need to start bunch of stuff, like compiler, debugger and so on. It leads to redundancy in this class. Try to move this part to another class(even Main).